backend/src/main/java/com/htecgroup/skynest/service/impl/FolderServiceImpl.java

```diff
@@ -136,6 +138,33 @@ public List<FolderResponse> getAllFoldersWithParent(UUID parentFolderId) {
     return asFolderResponseList(allFolders);
   }
 
+  @Override
+  public void moveFolderToBucket(UUID folderId) {
+    FolderEntity folderEntity =
+        folderRepository.findById(folderId).orElseThrow(FolderNotFoundException::new);
+    if (folderEntity.getParentFolder() == null) {
+      throw new FolderAlreadyInsideBucketException();
+    }
+    folderEntity.setParentFolder(null);
+    folderRepository.save(folderEntity);
+    actionService.recordAction(Collections.singleton(folderEntity), ActionType.MOVE);
+  }
+
+  @Override
+  public void moveFolderToFolder(UUID folderId, UUID destinationFolderId) {
+    FolderEntity folderEntity =
+        folderRepository.findById(folderId).orElseThrow(FolderNotFoundException::new);
+    FolderEntity parentFolderEntity =
+        folderRepository.findById(destinationFolderId).orElseThrow(FolderNotFoundException::new);
```

Comment:
why parentFolderEntity? shouldn't it be destinationFolderEntity?

Reply:
but it'll be its parent folder if you move it, I guess both are right.